client/FxAccountClient.js

```diff
@@ -1031,4 +1099,30 @@ define([
   return FxAccountClient;
 });
 
+var METRICS_CONTEXT_OPTIONS = {
+  context: true,
+  entrypoint: true,
+  migration: true,
+  service: true,
+  utmCampaign: true,
+  utmContent: true,
+  utmMedium: true,
+  utmSource: true,
+  utmTerm: true
+};
+
+function marshallMetricsContext (data) {
```

Comment:
Ha, I used to be a COM guy once upon a time, so I think that's why it came to me.